src/polyline2_signed_distance_constraint.cpp

```diff
@@ -58,9 +58,12 @@ bool Polyline2SignedDistanceConstraint::IsSatisfied(const VectorXf& input,
 
   // Compute signed squared distance by finding closest point.
   float signed_distance_sq;
+  bool is_endpoint;
   polyline_.ClosestPoint(Point2(input(xidx_), input(yidx_)), nullptr, nullptr,
-                         &signed_distance_sq);
-
+                         &signed_distance_sq, &is_endpoint);
+  if(is_endpoint){
+    signed_distance_sq = 0;
+  }
   // Maybe set level.
```

Comment:
nit: blank line above this

Reply:
done!

Reply:
See comment above about not seeing change here.

Reply:
On second thought, just disregard this comment